docs/how-to-guides/contributing.md

```diff
@@ -0,0 +1,33 @@
+This document explains the processes and practices recommended for contributing enhancements to the Indico operator.
+
+* Generally, before developing enhancements to this charm, you should consider [opening an issue ](https://github.com/canonical/discourse-k8s-operator/issues)explaining your use case.
+* If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
+* Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+* All enhancements require review before being merged. Code review typically examines
+  * code quality
+  * test coverage
+  * user experience for Juju administrators of this charm.
+* Please help us out in ensuring easy to review branches by rebasing your pull request branch onto the `main` branch. This also avoids merge commits and creates a linear Git commit history.
```

Comment:
We ensure a linear commit history using squash merges, no? In that case what devs do in their branch doesn't really matter, right?

Reply:
Indeed. Same as above, we'll have to remove this paragraph in for other charms